shared/haml/user_header.haml

```diff
@@ -0,0 +1,95 @@
+-# Locals: current_user, loc_prefix, [session_pairings]
+- if current_user
+  .header_button.header_user.user_menu
+    - if current_user.can_pair? && session_pairings.present?
+      %i.fa.fa-users
+      %span= I18n.t("#{loc_prefix}label", username: I18n.t("#{loc_prefix}team"))
+    - else
+      - short_name = ERB::Util.h(current_user.short_name)
+      %span.user_name{'data-id': current_user.id, 'data-shortname': short_name}= I18n.t("#{loc_prefix}label", username: short_name)
+    %span.user_menu_glyph
+      &#x25BC;
+    %div{style: 'position: relative; top: 3px;'}
+      .user_options{style: 'display: none;'}
+        - if current_user.can_pair?
+          - if session_pairings.present?
+            = link_to '#', {id: 'pairing_link', style: 'display: none'} do
+              %i.fa.fa-user-plus
+              = I18n.t("#{loc_prefix}pair_programming")
+              .pairing_summary
+                #{I18n.t("#{loc_prefix}driver")}:
+                = h(current_user.short_name)
+                - session_pairings.map do |id|
+                  %br
+                  #{I18n.t("#{loc_prefix}navigator")}:
+                  = h(User.find(id).short_name)
+          - else
+            = link_to '#', {id: 'pairing_link', style: 'display: none'} do
+              %i.fa.fa-user-plus
+              = I18n.t("#{loc_prefix}pair_programming")
+        %a{href: CDO.studio_url('users/edit')}= I18n.t("#{loc_prefix}settings")
+        %a{href: CDO.studio_url('users/sign_out')}= I18n.t("#{loc_prefix}logout")
+- else
+  %a{href: CDO.studio_url('users/sign_in'), class: 'button-signin', id: 'signin_button'}
+    .header_button.header_user
+      %span= I18n.t("#{loc_prefix}signin")
+:javascript
+  // Use function closure to avoid cluttering global namespace.
+  (function() {
+    var cookieKey = '#{'_shortName' + (rack_env?(:production) ? '' : "_#{rack_env}")}';
+    // Share cookie with top-level domain.
+    var topLevelDomain = document.location.hostname.split('.').slice(-2).join('.');
+    var cookieMeta = "path=/;domain=" + topLevelDomain;
+
+    // Provide current_user.short_name to cached pages via session cookie.
+    var nameSpan = document.querySelector('.header_button.header_user.user_menu .user_name');
+    if (nameSpan) {
+      var dataName = nameSpan.dataset.shortname;
+      var id = nameSpan.dataset.id;
+      var cookieMatch = document.cookie.match('(^|;) ?' + cookieKey + '=([^;]*)(;|$)');
+      var storedName = cookieMatch && cookieMatch[2];
+      if (storedName && !id) {
+        nameSpan.innerHTML = nameSpan.innerHTML.replace(dataName, storedName);
+      } else if (!storedName && id) {
+        document.cookie = cookieKey + '=' + dataName + ';' + cookieMeta;
+      }
+    }
+
+    function hideUserOptions() {
+      $('.user_options').slideUp();
+      $('.user_menu_glyph').html('&#x25BC;');
```

Comment:
Actually if this change will affect a whole bunch of eyes tests anyway, it would be a _great_ time to switch over to using the same font-awesome up/down glyphs used inside the hamburger.  (If not, then don't worry.. I just know there will be a git eyes hit when we finally do switch to using font-awesome.)